cmd/kubeadm/app/cmd/phases/markcontrolplane.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package phases
+
+import (
+	"path/filepath"
+	
+	"github.com/spf13/cobra"
+	"github.com/pkg/errors"
+	
+	kubeadmscheme "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/scheme"
+	kubeadmapiv1beta1 "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1beta1"
+	"k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/validation"
+	"k8s.io/kubernetes/cmd/kubeadm/app/cmd/options"
+	cmdutil "k8s.io/kubernetes/cmd/kubeadm/app/cmd/util"
+	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
+	"k8s.io/kubernetes/cmd/kubeadm/app/cmd/phases/workflow"
+	markcontrolplanephase "k8s.io/kubernetes/cmd/kubeadm/app/phases/markcontrolplane"
+	kubeadmutil "k8s.io/kubernetes/cmd/kubeadm/app/util"
+	configutil "k8s.io/kubernetes/cmd/kubeadm/app/util/config"
+	kubeconfigutil "k8s.io/kubernetes/cmd/kubeadm/app/util/kubeconfig"
+	"k8s.io/kubernetes/pkg/util/normalizer"
+)
+
+var (
+	
+	markControlPlaneExample = normalizer.Examples(`
+		# Applies control-plane label and taint to the current node, functionally equivalent to what is executed by kubeadm init.
+		kubeadm init phase mark-control-plane --config config.yml
+
+		# Applies master label and taint to a specific node
+		kubeadm init phase mark-control-plane --node-name myNode
+		`)
+)
+
+
+type markControlPlaneData interface {
+	Cfg() *kubeadmapi.InitConfiguration
+	KubeConfigDir() string
+	DryRun() bool
+}
+
+// NewMarkControlPlanePhase creates a kubeadm workflow phase that implements mark-controlplane checks.
+func NewMarkControlPlanePhase() workflow.Phase {
+	return workflow.Phase{
+		Name:    "mark-control-plane",
+		Short:   "Mark a node as a control-plane",
+		Example: markControlPlaneExample,
+		Run:     runMarkControlPlane,
+	}
+}
+
+// runMarkControlPlane executes markcontrolplane checks logic.
+func runMarkControlPlane(c workflow.RunData) error {
+	data, ok := c.(markControlPlaneData)
+	if !ok {
+		return errors.New("mark-control-plane phase invoked with an invalid data struct")
+	}
+	
+	var cfgPath string
+	kubeConfigFile := filepath.Join(data.KubeConfig(), kubeadmconstants.AdminKubeConfigFileName)
+	kubeConfigFile = cmdutil.FindExistingKubeConfig(kubeConfigFile)
```

Comment:
The call to `FindExistingKubeConfig` should be removed and instead we should use data.Kubeconfig

@neolit123 the call of  `FindExistingKubeConfig` should be removed by all phases, because it can make kubeadm use unexpected kubeconfig files while instead we need admin.conf.

Reply:
^ started a thread on slack.

Reply:
ok we had a discussion about this - @RA489 please remove the FindExistingKubeConfig call and use:
`kubeConfigFile := data.KubeConfigPath()`

Reply:
@neolit123 ok